packages/styling/src/utilities/icons.ts

```diff
@@ -121,19 +139,21 @@ export function getIcon(name?: string): IIconRecord | undefined {
 
     if (icon) {
       let { subset } = icon;
+      if (subset && subset.fontFace) {
 
-      if (subset.fontFace && !subset.isRegistered) {
-        // Register font face for given icons.
-        fontFace(subset.fontFace);
-
-        // Generate a base class name for the given font.
-        subset.className = mergeStyles(subset.style, {
-          fontFamily: subset.fontFace.fontFamily,
-          fontWeight: subset.fontFace.fontWeight || 'normal',
-          fontStyle: subset.fontFace.fontStyle || 'normal'
-        }).toString();
+        if (!subset.isRegistered) {
+          fontFace(subset.fontFace);
+          subset.isRegistered = true;
+        }
 
-        subset.isRegistered = true;
+        subset.className = mergeStyles(
```

Comment:
This mergeStyles call used to be done only once for a subset, now it would be done on every getIcon call.

I think this could be tweaked to do this once again but reset only the mergeStyles part on stylesheet reset.